### ISSUE TITLE
Asynchronous retrieval help

### ISSUE THREAD

**Issue by ahaverty** [nmoskalenko/rxFirebase#7]
This isn't particularly an issue with RxFirebase but I'm wondering if its currently possible. @renanferrari perhaps you may be able to answer this also?

In some cases I want to asynchronously return a value. I've tried a fair few suggestions on RxJava forums including stuff like:

``` java
User user = RxFirebaseDatabase.observeSingleValue(
    reference.child("users").child("nick"), User.class).toBlocking().single();
```

But haven't had any luck.
Have you any suggestions on how to do so, or even synchronously wait for multiple observers, so as to avoid long chains of the following:

```
// My current anti-pattern
#
call1.subscribe(
    call2.subscribe(
        call3.subscribe(
            //Now do something with all 3 pieces of data
        )
    )
)
```

Thank you

**Comment by renanferrari** [comment]
Hey @ahaverty, I'm not sure if I understood your question. So let me try to clarify it a little bit:

In the first example it looks like you want to get a single User. But the way you put it, by declaring "User user = ..." and ".toBlocking()", makes me think you want to get it in a synchronous way, not asynchronous. Is that right? Do you want to block everything else happening in your app until your method have returned an User object?

In the second example I think it got a little bit clearer for me where you want to go with this. Maybe you want to read about combining multiple Observables [here](https://github.com/ReactiveX/RxJava/wiki/Combining-Observables).

Let me know if I can help you with anything else.

**Comment by ahaverty** [comment]
Hi @renanferrari,
I spent the last 4 years of college understanding 'Synchronously' to mean 'concurrently'.
6 weeks ago, starting with firebase for the first time, I came across the google definition of Asynchronously = 'not existing or occurring at the same time', which flipped my mind for a day. Since then I have taken it that asynchronously meant single thread/waiting for each process to complete.

After reading your comment, and a quick SO search, you've brought back my sanity: http://stackoverflow.com/questions/748175/asynchronous-vs-synchronous-execution-what-does-it-really-mean
(^Atleast I'm not alone by the looks of the 29 upvotes in the comments.)

It now makes sense why I got nowhere searching RxJava Asynch.
Thanks for the reality check and link!
